tsdb/head.go

```diff
@@ -1042,15 +1042,17 @@ func (a *headAppender) log() error {
 }
 
 func (a *headAppender) Commit() error {
+	if err := a.log(); err != nil {
+		//nolint: errcheck
+		a.Rollback() // Most likely the same error will happen again.
```

Comment:
`a.head.putSeriesBuffer(a.sampleSeries)` seems to be missing in `Rollback` (which is in `Commit`). Not sure if that intentional, but I guess we could add it in there too.